src/parser/Parser.ts

```diff
@@ -1199,6 +1197,13 @@ export class Parser {
             });
             throw this.lastDiagnosticAsError();
         }
+        let itemType = new DynamicType();
```

Comment:
Moved the typing of the for-each item to BEFORE the parsing of the inside, because there could be an inside loop that was iterating over some property of the outside item